### PR TITLE
schema-reporting: Options renaming + remove experimental_ prefix

### PR DESCRIPTION
**Pull request by sachindshinde**
This PR removes the `experimental_` prefix from schema reporting option names, and specifically changes `experimental_schemaReporting` to `reportSchema`. Note that the old option names are still there, but they've been marked `@deprecated` and trying to use them will log a warning.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ The version headers in this history reflect the versions of Apollo Server itself
...
 - `apollo-server-core`: Though Apollo Server **is not affected** due to the way it is integrated, in response to [an upstream security advisory for GraphQL Playground](https://github.com/prisma-labs/graphql-playground/security/advisories/GHSA-4852-vrh7-28rf) we have published [the same patch](https://github.com/prisma-labs/graphql-playground/commit/bf1883db538c97b076801a60677733816cb3cfb7) on our `@apollographql/graphql-playground-html` fork and bumped Apollo Server to use it.  Again, this was done out of an **abundance of caution** since the way that Apollo Server utilizes `renderPlaygroundPage` is _not_ vulnerable as it does not allow per-request Playground configuration that could allow interpolation of user-input. [PR #4231](https://github.com/apollographql/apollo-server/pull/4231)
+- `apollo-engine-reporting`: Change the `experimental_schemaReporting` option name to `experimental_reportSchema`. [PR #4236](https://github.com/apollographql/apollo-server/pull/4236)
```
Comment (glasser) [inline CHANGELOG.md:15]: Surely this shouldn't contain experimental?

Comment (sachindshinde) [reply CHANGELOG.md]: Sorry for the confusion, I made two changelog entries here when really I should have just condensed them into one.

**Comment by sachindshinde** [comment]
(I'll update this PR once #4244 and #4246 merge)

**Comment by jsegaran** [comment]
I also think you will want to add a line to `docs/source/api/apollo-server.md` It looks like I didn't add one for the experimental flags. i think we can just add the new fields?

**Comment by abernix** [comment]
I've changed the target branch to `release-2.15.0`.  We can cut an alpha of this from that release branch before the final version is published.

**Pull request merged**